client/src/components/Reviews/components/AddReview.jsx

```diff
@@ -1,2 +1,203 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
+import { FaRegStar, FaStar } from 'react-icons/fa';
+import axios from 'axios';
+
+// TODO Change this to be the modal styles to make it a popup window
+const Modal = styled.div`
+  position: relative;
+  border: 2px solid black;
+  color: black;
+  width: calc(60% - (.5em + 6px));
+  float: right;
+  min-height: 400px;
+  margin-top: 1em;
+`;
+const HiddenRadioButton = styled.input.attrs({
+  type: 'radio',
+})`
+  height: 25px;
+  width: 25px;
+  cursor: pointer;
+  position: absolute;
+  opacity: 0;
+`;
+
+const data = {
+  Size: ['A size too small', '1/2 a size too small', 'Perfect', '1/2 a size too big', 'A size too big'],
+  Width: ['Too narrow', 'Slightly narrow', 'Perfect', 'Slightly wide', 'Too wide'],
+  Comfort: ['Uncomfortable', 'Slightly uncomfortable', 'Ok', 'Comfortable', 'Perfect'],
+  Quality: ['Poor', 'Bewlow average', 'What I expected', 'Pretty great', 'Perfect'],
+  Length: ['Runs Short', 'Runs slightly short', 'Perfect', 'Runs slightly long', 'Runs long'],
+  Fit: ['Runs tight', 'Runs slightly tight', 'Perfect', 'Runs slightly long', 'Runs long'],
+};
+const starsMeaning = ['Poor', 'Fair', 'Average', 'Good', 'Great'];
+
+function AddReview({ handleAddReview, chars }) {
+  const [recommended, setRecommended] = useState(null);
+  const [rating, setRating] = useState(null);
+  const [img, setImg] = useState(null);
+  const [reviewText, setReviewText] = useState({
+    summary: '',
+    body: '',
+    name: '',
+    email: '',
+  });
+  const [characteristics, setCharacteristics] = useState({
+    Size: null,
+    Width: null,
+    Comfort: null,
+    Quality: null,
+    Length: null,
+    Fit: null,
+  });
+
+  function handelReviewSubmit(e) {
+    // TODO add in checks for data
+    // TODO finish data submisson
+    e.preventDefault();
+    const submitedData = {
+      ...reviewText,
+      product_id: 'NEED TO CHANGE', // TODO get product id
```

Comment:
Maybe take `product_id` as a prop to your components?

Reply:
I take `product_id` as a prop in `Overview`. In my mind, the main app (i.e. the product detail page) tells us what product is being displayed through `product_id` and then each of the component can fetch the data they need. Unless there is some shared data that we can fetch once in the top level component for efficiency.